pkg/columns/formatter/json/json.go

```diff
@@ -208,34 +207,79 @@ func (f *Formatter[T]) getPrinter(level int, cols []*column[T]) func(entry *T, b
 		}
 		handlePrefix()
 
-		funcs = append(funcs, func(entry *T, buf *encodeState) {
+		funcs = append(funcs, func(buf *encodeState, entry *T, indent string) {
 			col.formatter(buf, entry)
 		})
 	}
 
-	return func(entry *T, buf *encodeState) {
+	return func(buf *encodeState, entry *T, indent string) {
 		for _, f := range funcs {
-			f(entry, buf)
+			f(buf, entry, indent)
 		}
 	}
 }
 
+func (f *Formatter[T]) formatEntry(buf *encodeState, entry *T, indent string) {
+	if entry == nil {
+		buf.WriteString(indent + "null")
+		return
+	}
+	buf.WriteString(indent + "{")
+	f.printer(buf, entry, indent)
+	if f.options.prettyPrint {
+		buf.WriteString("\n" + indent)
+	}
+	buf.WriteByte('}')
+}
+
 // FormatEntry returns an entry as a formatted string, respecting the given formatting settings
 func (f *Formatter[T]) FormatEntry(entry *T) string {
-	if entry == nil {
-		return ""
+	buf := bufpool.Get().(*encodeState)
+	buf.Reset()
+	defer bufpool.Put(buf)
+
+	f.formatEntry(buf, entry, "")
+
+	return buf.String()
+}
+
+// FormatEntries returns a slice of entries as a formatted string, respecting the given formatting settings
+func (f *Formatter[T]) FormatEntries(entries []*T) string {
+	if entries == nil {
+		return "null"
+	}
+
+	// TODO[Mauricio]: I can't remember why but the behavior of the default golang formatter was
+	// causing issues when the slice was empty. So let's marshall to [] instead of null on this case.
+	if len(entries) == 0 {
+		return "[]"
 	}
 
 	buf := bufpool.Get().(*encodeState)
 	buf.Reset()
 	defer bufpool.Put(buf)
 
-	buf.WriteByte('{')
-	f.printer(entry, buf)
-	if f.options.prettyPrint {
-		buf.WriteByte('\n')
+	if !f.options.prettyPrint {
+		buf.WriteByte('[')
+		for l, entry := range entries {
+			f.formatEntry(buf, entry, "")
+			if l < len(entries)-1 {
+				buf.WriteString(", ")
+			}
+		}
+		buf.WriteByte(']')
+
+		return buf.String()
 	}
-	buf.WriteByte('}')
+
+	buf.WriteString("[\n")
```

Comment:
Maybe instead of a return in the `if` we can have a
```golang
} else {
 ...
}
return buf.String()
```

Normally I am also the person for early returns but this just looks in my head perfect for a if-else :sweat_smile:

Reply:
Right, makes sense on this case.